code/modules/events/crystal_event.dm

```diff
@@ -0,0 +1,412 @@
+		return
+	..()
+
+/obj/structure/crystal_portal/attackby(obj/item/W, mob/living/user, params)
+	if(!istype(W) || (W.item_flags & ABSTRACT) || !istype(user))
+		return
+	if(istype(W, /obj/item/anomaly_neutralizer))
+		to_chat(user, "<span class='notice'>You start closing the [src]</span>")
+		if(W.use_tool(src, user, 6.5 SECONDS, volume=100))
+			to_chat(user, "<span class='notice'>You you successfully close the [src]</span>")
+			qdel(src)
+
+
+/obj/structure/crystal_portal/small
+	name = "Small Portal"
+	desc = "A small portal to an unkown dimension!"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "anom"
+	color = "#B2FFFE"
+	anchored = TRUE
+	max_mobs = 3
+	spawn_time = 200
+	mob_types = list(/mob/living/simple_animal/hostile/crystal_monster/minion, /mob/living/simple_animal/hostile/crystal_monster/thug)
+	spawn_text = "emerges from"
+	faction = list("crystal")
+/obj/structure/crystal_portal/medium
+	name = "Medium Portal"
+	desc = "A medium portal to an unkown dimension!"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "anom"
+	color = "#93F3B2"
+	anchored = TRUE
+	max_mobs = 5
+	spawn_time = 180
+	mob_types = list(/mob/living/simple_animal/hostile/crystal_monster/minion, /mob/living/simple_animal/hostile/crystal_monster/thug,\
+					 /mob/living/simple_animal/hostile/crystal_monster/recruit)
+	spawn_text = "emerges from"
+	faction = list("crystal")
+/obj/structure/crystal_portal/big
+	name = "Big Portal"
+	desc = "A big portal to an unkown dimension!"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "anom"
+	color = "#F4F48A"
+	anchored = TRUE
+	max_mobs = 8
+	spawn_time = 160
+	mob_types = list(/mob/living/simple_animal/hostile/crystal_monster/minion, /mob/living/simple_animal/hostile/crystal_monster/thug,\
+					 /mob/living/simple_animal/hostile/crystal_monster/recruit, /mob/living/simple_animal/hostile/crystal_monster/killer)
+	spawn_text = "emerges from"
+	faction = list("crystal")
+/obj/structure/crystal_portal/huge
+	name = "Huge Portal"
+	desc = "A huge portal to an unkown dimension!"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "anom"
+	color = "#F97575"
+	anchored = TRUE
+	max_mobs = 12
+	spawn_time = 140
+	mob_types = list(/mob/living/simple_animal/hostile/crystal_monster/minion, /mob/living/simple_animal/hostile/crystal_monster/thug,\
+					 /mob/living/simple_animal/hostile/crystal_monster/recruit, /mob/living/simple_animal/hostile/crystal_monster/killer, \
+					 /mob/living/simple_animal/hostile/crystal_monster/boss)
+	spawn_text = "emerges from"
+	faction = list("crystal")
+
+/*
+This section is for the crystal monsters variations
+*/
+
+/mob/living/simple_animal/hostile/crystal_monster
+	name = "crystal monster"
+	desc = "A monster made of crystals similar to the Supermatter ones."
+	icon = 'icons/mob/simple_human.dmi'
+	icon_state = "crystal_minion"
+	icon_living = "crystal_minion"
+	icon_dead = "crystal_minion"
+	gender = NEUTER
+	mob_biotypes = MOB_UNDEAD|MOB_HUMANOID
```

Comment:
shouldn't they be inorganic, not undead?

Reply:
i mean they are skeletons with crystals on them

Reply:
Plasmamen are also skeletons but they are classified as inorganics. Your call though.